nunaliit2-couch-metadata/src/main/java/ca/carleton/gcrc/couch/metadata/IndexServlet.java

```diff
@@ -0,0 +1,240 @@
+            }
+        }
+
+        // Clone the HTML document so we don't modify shared copy.
+        Document indexHtml = indexDbChangeListener.getAttachment().clone();
+        if (indexHtml != null) {
+            response.setStatus(HttpServletResponse.SC_OK);
+            response.setContentType("text/html");
+
+            if (metadata != null) {
+                Element jsonLd = indexHtml.createElement("script");
+                jsonLd.attr("type", "application/ld+json");
+                jsonLd.text(metadata.toString());
+
+                indexHtml.head().appendChild(jsonLd);
+            }
+            else {
+                logger.debug("No metadata found for publishing");
+            }
+
+            try {
+                OutputStream os = response.getOutputStream();
+                os.write(indexHtml.html().getBytes());
+            }
+            catch (Exception e) {
+                logger.error("Problem outputting index.html", e);
+            }
+        }
+        else {
+            logger.error("No index.html found, returning {}", HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public void destroy() {
+        super.destroy();
+        logger.info("Destroyed");
+    }
+
+    /**
+     * Finds the metadata object associated with the module document.
+     *
+     * @param moduleDocId Module document Id to find metdata for.
+     * @return The JSON-LD document representing the metadata.
+     */
+    private JSONObject findModuleMetadata(String moduleDocId) {
+        return moduleMetadataChangeListener.findMetadata(moduleDocId);
+    }
+
+    /**
+     * Finds the atlas metadata document in the database. The atlas metadata document is identified by the fact that it
+     * is the only metadata document without a reference to a module.
+     *
+     * @return The JSON-LD document representing the metadata.
+     */
+    private JSONObject findAtlasMetadata() {
+        JSONObject atlasDocument = null;
+        JSONObject atlasMetadata = null;
+        try {
+            atlasDocument = couchDb.getDocument(CouchNunaliitConstants.ATLAS_DOC_ID);
+        }
+        catch (Exception e) {
+            logger.warn("Error accessing atlas metadata: {}", e.getMessage());
+        }
+
+        if (atlasDocument != null) {
+            JSONObject atlas = atlasDocument.optJSONObject("nunaliit_atlas");
+            if (atlas != null) {
+                // nunaliit_metadata contains the JSON-LD document to publish.
+                atlasMetadata = atlas.getJSONObject("nunaliit_metadata");
+            }
+        }
+        else {
+            logger.warn("Atlas document '{}' not found in database, cannot output metadata", CouchNunaliitConstants.ATLAS_DOC_ID);
+        }
+
+        return atlasMetadata;
+    }
+
+    /**
+     * Looks for the module document Id in the given query string.
+     * <p>
```

Comment:
Again are these html elements suppose to be here in the comments?

Reply:
They are supported per JavaDoc spec. Probably IntelliJ inserted them for me.